src/test/streams_tests.cpp

```diff
@@ -225,29 +285,34 @@ BOOST_AUTO_TEST_CASE(streams_serializedata_xor)
     // Degenerate case
     {
         DataStream ds{in};
-        ds.Xor({0x00, 0x00});
+        ds.Xor(0);
         BOOST_CHECK_EQUAL(""s, ds.str());
     }
 
     in.push_back(std::byte{0x0f});
     in.push_back(std::byte{0xf0});
 
-    // Single character key
     {
+        const std::vector xor_pat{std::byte{0xff}, std::byte{0xff}, std::byte{0xff}, std::byte{0xff}, std::byte{0xff}, std::byte{0xff}, std::byte{0xff}, std::byte{0xff}};
+        uint64_t xor_key;
+        std::memcpy(&xor_key, xor_pat.data(), 8);
+
         DataStream ds{in};
-        ds.Xor({0xff});
+        ds.Xor(xor_key);
         BOOST_CHECK_EQUAL("\xf0\x0f"s, ds.str());
     }
 
-    // Multi character key
-
     in.clear();
     in.push_back(std::byte{0xf0});
     in.push_back(std::byte{0x0f});
 
     {
+        const std::vector xor_pat{std::byte{0xff}, std::byte{0x0f}, std::byte{0xff}, std::byte{0x0f}, std::byte{0xff}, std::byte{0x0f}, std::byte{0xff}, std::byte{0x0f}};
+        uint64_t xor_key;
+        std::memcpy(&xor_key, xor_pat.data(), 8);
+
         DataStream ds{in};
-        ds.Xor({0xff, 0x0f});
+        ds.Xor(xor_key);
```

Comment:
It doesn't (at least not for all test cases), I'm deliberately testing generating vectors and converting them instead of generating 64 bit values since that's what happens in production code (this should address multiple of your comments)

Reply:
[My bad](https://github.com/bitcoin/bitcoin/pull/31144#discussion_r1837275100).